var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -85,6 +85,13 @@ class Hypre(AutotoolsPackage, CudaPackage, ROCmPackage):
     variant("caliper", default=False, description="Enable Caliper support")
     variant("rocblas", default=False, description="Enable rocBLAS")
     variant("cublas", default=False, description="Enable cuBLAS")
+    variant(
+        "precision",
+        default="double",
+        values=("single", "double", "longdouble"),
+        multi=False,
+        description="Floating point precision",
+    )
```

Comment:
I think it has been available in hypre for a long time, just the package did not have it as a variant.

I checked -- the precision options were added in hypre v2.12.1: `--enable-single` and `--enable-longdouble`.

I though all `hypre` versions supported by the package have these but it turns out some of the oldest ones don't. Do you think it is worth adding the constraint `when="@2.12.1:"`?

Reply:
> Do you think it is worth adding the constraint when="@2.12.1:"?

Since there are some listed `hypre` versions before this, that would be useful (although I acknowledge that it's unlikely folks will be building these old versions, we try not to get in the way of doing so).

Reply:
Addressed in 1eaf8a663e.